subprojects/packagefiles/iir1/test/meson.build

```diff
@@ -0,0 +1,15 @@
+tests = [
+  { 'name' : 'badparam',    'should_fail' : true }, 
+  { 'name' : 'butterworth', 'should_fail' : false },
+  { 'name' : 'chebyshev1',  'should_fail' : false },
+  { 'name' : 'chebyshev2',  'should_fail' : false },
+  { 'name' : 'custom',      'should_fail' : false },
+  { 'name' : 'rbj',         'should_fail' : false },
+  { 'name' : 'state',       'should_fail' : false },
+]
+
+foreach t : tests
+  name = t.get('name')
+  exe = executable(name, name + '.cpp', dependencies : libiir1_dep)
+  test(name, exe, should_fail : t.get('should_fail'))
```

Comment:
I haven't investigated, but upstream lists all these as ordinary cmake add_tests, can you clarify why some of them are should_fail?

Reply:
>  ordinary cmake add_tests

I use meson to avoid everything to do with cmake, so I'm not sure what these are.

If there's a more correct way of running a self test, in these cases, beyond the "test(...)" call, then I'm happy to be pointed to that and will adjust.

Reply:
What I mean is I don't see anywhere in the cmake build description that says this command is supposed to return non-zero.

So I got a chance to try this wrap locally.

```
$ ninja -C builddir test
ninja: Entering directory `builddir'
[0/1] Running all tests.
1/7 badparam           UNEXPECTEDPASS  0.02s
>>> MALLOC_PERTURB_=230 /home/eschwartz/git/mesonbuild/wrapdb/builddir/subprojects/iir1-1.9.2/test/badparam
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stderr:
Correct lowpass exception thrown for fc = fs/2: The cutoff frequency needs to be below the Nyquist frequency.
Correct highpass exception thrown for fc = fs/2: The cutoff frequency needs to be below the Nyquist frequency.
Correct bandstop exception thrown for fc = fs/2: The cutoff frequency needs to be below the Nyquist frequency.
Correct bandpass exception thrown for fc = fs/2: The cutoff frequency needs to be below the Nyquist frequency.
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

2/7 chebyshev2         OK              0.01s
3/7 custom             OK              0.01s
4/7 chebyshev1         OK              0.03s
5/7 rbj                OK              0.02s
6/7 state              OK              0.01s
7/7 butterworth        OK              0.44s

Summary of Failures:

1/7 badparam    UNEXPECTEDPASS  0.02s


Ok:                 6   
Expected Fail:      0   
Fail:               0   
Unexpected Pass:    1   
Skipped:            0   
Timeout:            0   

Full log written to /home/eschwartz/git/mesonbuild/wrapdb/builddir/meson-logs/testlog.txt
FAILED: meson-test 
/home/eschwartz/.local/bin/meson test --no-rebuild --print-errorlogs
ninja: build stopped: subcommand failed.
```

The should_fail is wrong, `test/badparam.cpp` uses the following pattern:
```cxx
        try {
                Iir::Butterworth::LowPass<3> f;
                f.setup (samplingrate, samplingrate / 2);
                assert_print(0,"No exception thrown by LowPass.");
        } catch ( const std::invalid_argument& e ) {
                fprintf(stderr,"Correct lowpass exception thrown for fc = fs/2: %s\n",e.what());
        }
```

> I use meson to avoid everything to do with cmake, so I'm not sure what and "ordinary cmake add_test" is versus other cmake non-ordinary or non-add_tests.

I use meson to avoid cmake too, but sadly I still have to know some cmake. :D I would expect that if a test is expected to return non-zero, Meson would have `should_fail: true` as a kwarg on the test, and cmake would have this sometime after the test to modify the test:
```cmake
set_tests_properties(testname PROPERTIES WILL_FAIL true)
```

Reply:
The reason this isn't picked up in the WrapDB CI is because the tests are disabled by default, and aren't re-enabled in ci_config.json

Reply:
Thanks for that correction; fixed.